CRM/Utils/SQL/TempTable.php

```diff
@@ -147,8 +147,7 @@ public function getUtf8String() {
     if (!$this->utf8) {
       return '';
     }
-    $dbUTF = CRM_Core_BAO_SchemaHandler::getDBCollation();
-    if (strpos($dbUTF, 'utf8') !== FALSE) {
+    if (CRM_Core_BAO_SchemaHandler::getDBCollation() === CRM_Core_BAO_SchemaHandler::getInUseCollation()) {
       return '';
```

Comment:
Thanks just this could still create a mismatch, e.g. if the db collation is utf8_general_ci and the table is using utf8mb4_0900_ai_ci, this would then return utf8_unicode_ci. I don't see a reason to use the db collation at all since the goal is to use the same as the existing tables. The only gotcha is civicrm_contact might be using a rogue collation different from other tables but that seems less likely than a db vs table mismatch. So something like:

```
$collation = CRM_Core_BAO_SchemaHandler::getInUseCollation();
if (empty($collation)) {
  return self::UTF8;
}
else {
  // get the charset from the collation
  $collation_parts = explode('_', $collation);
  return "DEFAULT CHARACTER SET {$collation_parts[0]} COLLATE {$collation}";
}
```